src/Sulu/Bundle/MediaBundle/Entity/MediaDataProviderRepository.php

```diff
@@ -0,0 +1,184 @@
+            ->leftJoin($alias . '.files', 'file')
+            ->leftJoin('file.fileVersions', 'fileVersion', 'WITH', 'fileVersion.version = file.version')
+            ->leftJoin('fileVersion.tags', 'tag')
+            ->leftJoin('fileVersion.meta', 'fileVersionMeta')
+            ->leftJoin('fileVersion.defaultMeta', 'fileVersionDefaultMeta')
+            ->leftJoin('fileVersion.contentLanguages', 'fileVersionContentLanguage')
+            ->leftJoin('fileVersion.publishLanguages', 'fileVersionPublishLanguage')
+            ->leftJoin($alias . '.creator', 'creator')
+            ->leftJoin('creator.contact', 'creatorContact')
+            ->leftJoin($alias . '.changer', 'changer')
+            ->leftJoin('changer.contact', 'changerContact')
+            ->addSelect('type')
+            ->addSelect('collection')
+            ->addSelect('file')
+            ->addSelect('tag')
+            ->addSelect('fileVersion')
+            ->addSelect('fileVersionMeta')
+            ->addSelect('fileVersionDefaultMeta')
+            ->addSelect('fileVersionContentLanguage')
+            ->addSelect('fileVersionPublishLanguage')
+            ->addSelect('creator')
+            ->addSelect('changer')
+            ->addSelect('creatorContact')
+            ->addSelect('changerContact');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function append(QueryBuilder $queryBuilder, $alias, $locale, $options = [])
+    {
+        $parameter = [];
+
+        if (array_key_exists('mimetype', $options)) {
+            $queryBuilder
+                ->andWhere('fileVersion.mimeType = :mimeType');
+
+            $parameter['mimeType'] = $options['mimetype'];
+        }
+        if (array_key_exists('type', $options)) {
+            $queryBuilder
+                ->innerJoin($alias . '.type', 'type')
+                ->andWhere('type.name = :type');
+
+            $parameter['type'] = $options['type'];
+        }
+
+        return $parameter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function appendTagsRelation(QueryBuilder $queryBuilder, $alias)
+    {
+        $queryBuilder
+            ->innerJoin($alias . '.files', 'file')
+            ->innerJoin('file.fileVersions', 'fileVersion', 'WITH', 'fileVersion.version = file.version');
+
+        return 'fileVersion.tags';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function appendDatasource($datasource, $includeSubFolders, QueryBuilder $queryBuilder, $alias)
+    {
+        if (!$includeSubFolders) {
+            $queryBuilder
+                ->innerJoin($alias . '.collection', 'collection')
+                ->andWhere('collection.id = :collectionId');
+        } else {
+            $queryBuilder
+                ->innerJoin(
+                    $this->collectionEntityName,
+                    'parentCollection',
+                    Join::WITH,
+                    'parentCollection.id = :collectionId'
+                )
+                ->innerJoin(
```

Comment:
Is this join really necessary? Do you really load data here, or is it just to restrict the result set? If it is the latter, I would prefer a where statement instead.